pkg/expr/expr_types.go

```diff
@@ -404,6 +432,85 @@ func CastToKV(val interface{}) (out map[string]string, err error) {
 	}
 }
 
+func (t *KV) Each(fn func(k string, v TypedValue) error) (err error) {
+	if t == nil || t.value == nil {
+		return
+	}
+
+	for k, v := range t.value {
+		if err = fn(k, Must(NewString(v))); err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+// Set set/update the specific key value in KV
+func (t *KV) Set(key string, val interface{}) error {
+	if t.value == nil {
+		t.value = make(map[string]string)
+	}
+
+	str, err := cast.ToStringE(UntypedValue(val))
+	t.value[key] = str
+
+	return err
+}
+
+// Merge combines the given KVs into KV
+func (t *KV) Merge(nn ...Iterator) (err error) {
+	if t.value == nil {
+		t.value = make(map[string]string)
+	}
+
+	nn = append([]Iterator{t}, nn...)
+
+	for _, i := range nn {
+		if err = i.Each(func(k string, v TypedValue) error {
+			t.value[k] = cast.ToString(v.Get())
```

Comment:
you can use `cast.ToStringE` to get the fn variant that does not suppress your errors